### PR TITLE
synapse rpc bittorrent integration

### PR DESCRIPTION
**Pull request by efrisch**
Initial integration with synapse closes pyrsia/pyrsia#199

### CODE REVIEW

File: src/network/torrent.rs [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+    let resources = if let SMessage::UpdateResources { resources, .. } = c.rr(msg)? {
+        resources
+    } else {
+        bail!("Failed to received torrent resource list!");
+    };
+
+    c.send(unsub)?;
+
+    let mut results = Vec::new();
```
Comment (mgrand) [inline src/network/torrent.rs:143]: Did you remember to run clippy?

**Review by mgrand — DISMISSED** [review]

**Comment by prince-chrismc** [comment]
Pretty please fix the link to your issue

 https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue

GitHub is super picky unfortunately for us

File: src/network/torrent.rs [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+}
+
+pub fn get_(c: &mut Client, id: &str, output: &str) -> Result<()> {
+    let res = get_resources(c, vec![id.to_owned()])?;
+    if res.is_empty() {
+        bail!("Resource not found");
+    }
+    match output {
+        "text" => {
+            println!("{}", res[0]);
```
Comment (MitaliBo) [inline src/network/torrent.rs:112]: can we change all the println with appropriate loggers

Comment (efrisch) [reply src/network/torrent.rs]: @MitaliBo Good catch! Done.

**Review by mgrand — APPROVED** [review]

**Review by MitaliBo — APPROVED** [review]

**Pull request merged**